src/modalities/models/coca/multi_modal_decoder.py

```diff
@@ -0,0 +1,123 @@
+from functools import partial
+from typing import Dict
+
+import torch
+import xformers.ops as xops
+from torch import nn
+
+from modalities.models.gpt2.gpt2_model import ActivationType, LayerNorm
+from modalities.models.model import NNModel
+from modalities.nn.attention import AttentionConfig, AttentionType, MultiHeadAttention
+from modalities.nn.mlp import MLP
+
+
+class TransformerBlock(nn.Module):
+    def __init__(
+        self,
+        n_embd: int,
+        bias: bool,
+        epsilon: float,
+        activation: ActivationType,
+        n_head: int,
+        dropout: float,
+        ffn_hidden: int,
+        with_context: bool,
+        attention_type: AttentionType,
+        attention_config: AttentionConfig = None,
+        add_extra_mlp: bool = False,
+    ):
+        super().__init__()
+        self.with_context = with_context
+        self.add_extra_mlp = add_extra_mlp or not with_context

```

Comment:
```suggestion
        self.add_extra_mlp = add_extra_mlp
```

Reply:
The `not with_context` part is redundant, as it also appears in lines 45 & 63. Alternatively, one could leave line 31 unchanged and reduce lines 45 & 63 to `if self.add_extra_mlp:`. However, the disadvantage of that approach is that we generally would have `self.add_extra_mlp != add_extra_mlp`, which might be confusing.